File: DelphicMaxims/reverse_stims.js
#!/usr/bin/env node
/**
 * Reverse stimulus and response in Delphic Maxims JSON file.
 * Changes from: show maxim text → type number
 * To: show number → type maxim text
 */

const fs = require('fs');
const path = require('path');

// Read the original file
const inputFile = path.join(__dirname, 'DelphicMaximsStims.json');
const outputFile = path.join(__dirname, 'DelphicMaximsStims_Reversed.json');

const data = JSON.parse(fs.readFileSync(inputFile, 'utf-8'));

// Reverse each stimulus
for (const cluster of data.setspec.clusters) {
    for (const stim of cluster.stims) {
        // Swap the correctResponse and clozeText values
        const oldResponse = stim.response.correctResponse;
        const oldDisplay = stim.display.clozeText;

        // Reverse them
        stim.response.correctResponse = oldDisplay;
        stim.display.clozeText = oldResponse;
    }
}

// Write to new file
fs.writeFileSync(outputFile, JSON.stringify(data, null, 4), 'utf-8');

console.log('✓ Created DelphicMaximsStims_Reversed.json');
console.log('  - Display (stimulus): Now shows the number (1-147)');
console.log('  - Response (expected): Now expects the maxim text');

File: DelphicMaxims/reverse_stims.py
#!/usr/bin/env python3
"""
Reverse stimulus and response in Delphic Maxims JSON file.
Changes from: show maxim text → type number
To: show number → type maxim text
"""

import json

# Read the original file
with open('DelphicMaximsStims.json', 'r', encoding='utf-8') as f:
    data = json.load(f)

# Reverse each stimulus
for cluster in data['setspec']['clusters']:
    for stim in cluster['stims']:
        # Swap the correctResponse and clozeText values
        old_response = stim['response']['correctResponse']
        old_display = stim['display']['clozeText']

        # Reverse them
        stim['response']['correctResponse'] = old_display
        stim['display']['clozeText'] = old_response

# Write to new file
with open('DelphicMaximsStims_Reversed.json', 'w', encoding='utf-8') as f:
    json.dump(data, f, indent=4, ensure_ascii=False)

print("✓ Created DelphicMaximsStims_Reversed.json")
print("  - Display (stimulus): Now shows the number (1-147)")
print("  - Response (expected): Now expects the maxim text")
